index.ts

```diff
@@ -167,6 +167,16 @@ function isNodeKindExportAssignment(value: ts.Node): value is ts.ExportAssignmen
 
 export default function generate(options: Options): Promise<void> {
 
+  // validate options
+	if (options.name && options.main) {
+		if (options.main.indexOf(options.name + '/') !== 0) {
```

Comment:
Should probably document a bit about what scenario this is handling, otherwise no one is going to remember in a few months. :)